x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_inference/ml_inference_logic.ts

```diff
@@ -359,7 +349,7 @@ export const MLInferenceLogic = kea<
     },
     startTextExpansionModelSuccess: () => {
       // Refresh ML models list when the text expansion model is started
-      actions.makeMLModelsRequest(undefined);
+      actions.startPollingModels();
```

Comment:
I wonder if we should have a `refreshModels` action on `CachedFetchModlesApiLogic` instead of re-using `startPollingModels` to force an update.

I'm fine with this for now, but its an improvement we could consider, it may not be worth the overhead though.

Reply:
@TattdCodeMonkey Did you mean like [this](https://github.com/elastic/kibana/pull/176257/commits/26ff28765e32320d5297b5d4b79e102589fa001e)?

Reply:
@TattdCodeMonkey For my understanding - is the issue that this action happens outside the pipeline configuration screen (ELSER callout), and so it unnecessarily starts polling the models rather than refreshing them once?

Reply:
It's more a semantic quibble, that really isn't important. 

If I under the code correctly we are calling  `startPollingModels()` here to force a refresh the model list after an update instead of waiting for the next poll, correct? If thats the case it would be a little nicer to have an action that is descriptive of that intent vs re-using `startPollingModels()`. 

BUT calling `startPollingModels()` works, so is it worth the complexity of introducing another action that may require other special handing to not interfere with the existing polling? maybe not I could make the argument both ways. So feel free to ignore me this time if you want :)

Reply:
Gotcha - yeah, it might not be worth the effort. But let's revisit this once these components and logics have been refactored.